File: server/Entities/Report.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace server.Entities;

[Table("reports")]
public class Report
{
    [Column("id")]
    public long Id { get; set; }
    [Column("industrial_facility_id")]
    public long IndustrialFacilityId { get; set; }
    public IndustrialFacility IndustrialFacility { get; set; }
    [Column("pollution_id")]
    public long PollutionId { get; set; }
    public Pollution Pollution { get; set; }
    [Column("year")]
    public int Year { get; set; }
    [Column("volume")]
    public float Volume { get; set; }
    [Column("tax_rate")]
    public float TaxRate { get; set; }
    [Column("tax_type")]
    public string TaxType { get; set; }
    [Column("tax_amount")]
    public float? TaxAmount { get; set; }
}

File: client/src/app/models/Report.ts
export interface ReportDto{
    id: number;
    pollutionId: number;
    industrialFacilityId: number;
    year: number;
    volume: number;
    taxRate: number;
    taxType: string;
}

export interface FullReportDto{
    id: number;
    pollutionName: string;
    facilityName: string;
    year: number;
    volume: number;
    massFlowRate: number;
    emissionsLimit: number;
    taxRate: number;
    taxType: string;
    taxAmount: number;
}